### PR TITLE
test(hooks): add tests for cwd_tracking, time_awareness, and auto_confirm

### PR DESCRIPTION
**Pull request by TimeToBuildBob**
## Summary

- Add comprehensive test coverage for three hooks that previously had **zero tests**
- **26 new tests** covering: `cwd_tracking` (7), `time_awareness` (13), `auto_confirm` (6)
- All tests pass, mypy clean, ruff clean

### cwd_tracking (7 tests)
- Pre-execute stores CWD in ContextVar
- Post-execute detects directory changes and yields system messages
- Error handling when `os.getcwd()` fails
- No message when CWD unchanged or no stored state

### time_awareness (13 tests)
- `_get_next_milestone()`: all milestone boundaries (0, 1, 5, 10, 15, 20, 30+)
- `add_time_message()`: initialization, milestone triggering, duplicate prevention, hour formatting, hidden message flag

### auto_confirm (6 tests)
- Confirms all tool types (shell, python, save, patch, browser, unknown)
- Works with workspace and preview parameters

## Test plan
- [x] All 26 new tests pass locally
- [x] All 154 existing hook tests still pass
- [x] mypy clean
- [x] ruff clean

----

> [!IMPORTANT]
> Add comprehensive test coverage for `cwd_tracking`, `time_awareness`, and `auto_confirm` hooks with 26 new tests.
> 
>   - **Tests Added**:
>     - `test_cwd_tracking.py`: 7 tests for `cwd_tracking` hook, covering CWD storage, directory change detection, and error handling.
>     - `test_time_awareness.py`: 13 tests for `time_awareness` hook, covering milestone detection, message formatting, and duplicate prevention.
>     - `test_auto_confirm.py`: 6 tests for `auto_confirm` hook, confirming various tool types and handling workspace and preview parameters.
>   - **Behavior**:
>     - `cwd_tracking`: Pre-execute stores CWD, post-execute detects changes, handles `os.getcwd()` errors.
>     - `time_awareness`: `_get_next_milestone()` handles various time boundaries, `add_time_message()` manages initialization and milestone messages.
>     - `auto_confirm`: Confirms all tool types, works with workspace and preview parameters.
>   - **Test Plan**:
>     - All 26 new tests pass locally.
>     - All 154 existing hook tests still pass.
>     - `mypy` and `ruff` checks are clean.
> 
> <sup>This description was created by </sup>[Ellipsis](https://www.ellipsis.dev?ref=gptme%2Fgptme&utm_source=github&utm_medium=referral)<sup> for ebf5d24fa1b8b635d78c6a35bab4abe9121272db. You can [customize](https://app.ellipsis.dev/gptme/settings/summaries) this summary. It will automatically update as commits are pushed.</sup>

### CODE REVIEW

**Comment by TimeToBuildBob** [comment]
@greptileai review

File: gptme/hooks/tests/test_auto_confirm.py [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+from typing import cast
+
+from gptme.hooks.auto_confirm import auto_confirm_hook
+from gptme.hooks.confirm import ConfirmAction
+from gptme.tools.base import ToolUse
+
+
+def _make_tool_use(tool: str, content: str = "") -> ToolUse:
+    """Create a ToolUse for testing."""
+    return cast(ToolUse, type("ToolUse", (), {"tool": tool, "content": content})())
```
[the remark this answers is not in the document]
Comment (TimeToBuildBob) [reply gptme/hooks/tests/test_auto_confirm.py]: Fixed in 501cd8ab9 — now uses real `ToolUse(tool=tool, args=[], content=content)` instead of the dynamic `type()` hack. Removed unused `cast` import.
